Services/DeviceModels.cs

```diff
@@ -153,5 +181,36 @@ private bool CheckStockDeviceModelExistence(string id)
             return this.stockDeviceModels.GetList()
                 .Any(model => id.Equals(model.Id, StringComparison.InvariantCultureIgnoreCase));
         }
+
+        /// <summary>
+        /// Transforms property names from Json Object to "xxx.yyy.zzz" format
+        /// </summary>
+        private void PreparePropertyNames(HashSet<string> set, object obj, string prefix)
+        {
+            /* Sample conversion:
+             * from -> Device : {
+             *              Reported : Properties
+             *              }
+             *          
+             * to -> Device.Reported.Properties
+             */
```

Comment:
1. the comment doesn't reflect what the code is doing
2. the code is not adding the prefix

Reply:
done